### PR TITLE
adding distributed data timing

### PR DESCRIPTION
**Pull request by Loic-Ker**
Adding distributed data computation and analysis

### CODE REVIEW

File: reports/distributed_datas/make_plot.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import pickle
+
+core = [0,1,2,3,4,5,6,7,8]
```
Comment (alessandro-canevaro) [inline reports/distributed_datas/make_plot.py]: Would be more elegant to do `list(range(9))`

File: src/data/distributed_data_loading.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import pickle
+import time
+
+import numpy as np
+import torch
+
+core = [0, 1, 2, 3, 4, 5, 6, 7, 8]
```
Comment (alessandro-canevaro) [inline src/data/distributed_data_loading.py]: same here

**Review by alessandro-canevaro — COMMENTED** [review]
1) I think the num_worker shouldn't be a fixed number but should be dynamically assigned based on the number of CPU cores.
2) The output plot doesn't really make sense to me. I was expecting that with more num_workers the data loading time should decrease.

**Review by alessandro-canevaro — APPROVED** [review]

**Review by erlahrafnkels — APPROVED** [review]
Looks good - only thing is I don't really understand either what the plot is telling us 😅

**Pull request merged**